src/renderers/BaseTimedMediaRenderer.js

```diff
@@ -229,6 +232,96 @@ export default class BaseTimedMediaRenderer extends BaseRenderer {
         }
     }
 
+    _applyFadeAudioOutBehaviour(behaviour, callback) {
+        const { duration } = behaviour;
+        logger.info(`fading out audio for ${this._rendererId} over ${duration}s`);
+        this.fadeOut(duration, behaviour.targetVolume ?? 0);
+        callback();
+    }
+
+    _applyFadeAudioInBehaviour(behaviour, callback) {
+        const { duration } = behaviour;
+        logger.info(`fading in audio for ${this._rendererId} over ${duration}s`);
+        this.fadeIn(duration, behaviour.startVolume ?? 0);
+        callback();
+    }
+
+    cancelFade() {
+        if (this._fadeIntervalId) {
+            clearInterval(this._fadeIntervalId);
+            this._fadeIntervalId = null;
+        }
+    }
+
+    pauseFade() {
+        this._fadePaused = true;
+    }
+
+    resumeFade() {
+        this._fadePaused = false;
+    }
+
+    cannotFade() {
+        // TODO: need to check this for foreground fading
+        return this._playoutEngine instanceof iOSPlayoutEngine;
```

Comment:
Do we need to sort this before merge? Also for clarity, I'm presuming this is foreground element fading, as we already fade background audio in and out automatically. Need to clarify this behaviour in manual - pointing out it's for foreground only as background does it whether you want it or not.

Reply:
I'm not sure we can sort it.  It is foreground fading; doesn't iOS disallow programmatic volume control?